x-pack/filebeat/module/fortinet/firewall/ingest/pipeline.yml

```diff
@@ -220,6 +194,70 @@ processors:
 - pipeline:
     name: '{< IngestPipeline "utm" >}'
     if: "ctx.fortinet?.firewall?.type == 'utm' || ctx.fortinet?.firewall?.type == 'dns'"
+- rename:
+    field: fortinet.firewall.reason
+    target_field: event.reason
+    ignore_missing: true
+- rename:
+    field: fortinet.firewall.msg
+    target_field: message
+    ignore_missing: true
+- rename:
+    field: fortinet.firewall.proto
+    target_field: network.iana_number
+    ignore_missing: true
+- script:
+    lang: painless
+    ignore_failure: true
+    if: ctx?.network?.iana_number != null
+    source: |
+      if (ctx?.network == null) {
+          ctx.network = new HashMap();
+      }
+      Map options = new HashMap();
+      options['1'] = 'icmp';
+      options['2'] = 'igmp';
+      options['6'] = 'tcp';
+      options['17'] = 'udp';
+      options['58'] = 'ipv6-icmp';
+      if (options[ctx.network.iana_number] != null) {
+        ctx.network.transport = options[ctx.network.iana_number];
```

Comment:
what do you mean? Else what? can you provide an example of what you mean?

Reply:
I originally just had 5 set processors with if conditions and wanted to condense the functionality

Reply:
Sorry, I wasn't clear enough. I thought using simple `if-else` statement performs better than creating the `options` HashMap to implement the same mapping logic here. Because the HashMap is created for every single log message and garbage collected later.

Using a map to define mapping makes code readable, I agree with that. But I was wondering if using `if-else` would perform better. Especially in case millions of events are processed by this pipeline.

As a sample code, I created this ingest pipeline:
```
PUT _ingest/pipeline/script-test
{
  "processors": [
    {
      "script": {
        "source": """
def iana_number = ctx.network.iana_number;
if (iana_number != null) {
  if (iana_number == '1') {
    ctx.network.transport = 'icmp';
    
  } else if (iana_number == '2') {
    ctx.network.transport = 'igmp';
    
  } else if (iana_number == '6') {
    ctx.network.transport = 'tcp';
    
  } else if (iana_number == '17') {
    ctx.network.transport = 'udp';
    
  } else if (iana_number == '58') {
    ctx.network.transport = 'ipv6-icmp';
  }
}  
        """
      }
    }
  ]
}
```

And using the simulate endpoint, I confirmed it works the same:
```
POST _ingest/pipeline/script-test/_simulate
{"docs":[
  {"_source":{"network":{"iana_number":"1"}}},
  {"_source":{"network":{"iana_number":"2"}}},
  {"_source":{"network":{"iana_number":"6"}}},
  {"_source":{"network":{"iana_number":"17"}}},
  {"_source":{"network":{"iana_number":"58"}}},
  {"_source":{"network":{"iana_number":"999"}}}
]}
```

As for performance, I wrote a sample Java test to verify the difference.
The following code processes the two approaches 1 million times.

```java
public class LoopTest {

    private void execute(String name, Function<String, String> mapping) {
        long start = System.currentTimeMillis();
        for (int i = 0; i < 1_000_000; i++) {
            String v = i % 2 == 0 ? "a" : "b";
            String r = mapping.apply(v);
        }
        long duration = System.currentTimeMillis() - start;
        System.out.printf("%s %d\n", name, duration);
    }

    @Test
    public void useMap() {
        execute("useMap", v -> {
            Map<String, String> options = new HashMap<>();
            options.put("a", "1");
            options.put("b", "2");
            return options.get(v);
        });
    }

    @Test
    public void useIf() {
        execute("useIf", v -> {
            String r = null;
            if ("a".equals(v)) {
                r = "1";
            } else if ("b".equals(v)) {
                r = "2";
            }
            return r;
        });
    }
}
```

Result:
```
useMap 144
useIf 17
```

Reply:
Gotcha. Ya I'm not a Java guy so the whole garbage collection thing isn't my cup of tea.

Reply:
done